src/App.tsx

```diff
@@ -1,10 +1,57 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [statusFilter, setStatusFilter] = useState('All');
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        const fetchedTodos = await getTodos();
+
+        setTodos(fetchedTodos);
+      } catch {
+        setError('Unable to load todos');
+      }
+    };
+
+    fetchTodos();
+  }, []);
+
+  useEffect(() => {
+    if (error) {
+      const timer = setTimeout(() => {
+        setError(null);
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+
+    return undefined;
+  }, [error]);
+
+  const filteredTodos = todos.filter(todo => {
+    if (statusFilter === 'All') {
+      return true;
+    }
+
+    if (statusFilter === 'Active') {
+      return !todo.completed;
+    }
+
+    if (statusFilter === 'Completed') {
+      return todo.completed;
+    }
+
+    return true;
```

Comment:
use switch case here and mov filter under cases to avoid extra iteration in 'All' case